map_it_test.go

```diff
@@ -789,11 +794,61 @@ func TestMap_EntryNotifiedEvent(t *testing.T) {
 	})
 }
 
+func TestMap_EntryNotifiedEventWithAddListener(t *testing.T) {
+	it.MapTester(t, func(t *testing.T, m *hz.Map) {
+		const totalCallCount = int32(100)
+		callCount := int32(0)
+		subscriptionID, err := m.AddListener(context.Background(), hz.MapListener{
+			EntryAdded: func(event *hz.EntryNotified) {
+				if event.EventType == hz.EntryAdded {
+					atomic.AddInt32(&callCount, 1)
+				} else {
+					t.Fatalf("unexpected event type: %v", event.EventType)
+				}
+			},
+		}, true)
+		if err != nil {
+			t.Fatal(err)
+		}
+		for i := 0; i < int(totalCallCount); i++ {
+			key := fmt.Sprintf("key-%d", i)
+			value := fmt.Sprintf("value-%d", i)
+			it.MustValue(m.Put(context.Background(), key, value))
+		}
+		it.Eventually(t, func() bool {
+			return atomic.LoadInt32(&callCount) == totalCallCount
+		})
+		atomic.StoreInt32(&callCount, 0)
+
```

Comment:
I guess a comment here would be more appropriate here than a  blank line.

Reply:
Well, I went over this test again and did not understand why I put some entries to the map after removing the listener. Also, even if we do that the counter should not increase. I did not know how the test passes. Can you enlighten me if you understand the logic? If it is also weird to you I will remove this part:

```js
                // Clear the map to get entry added events
		err = m.Clear(context.Background())
		if err != nil {
			t.Fatal(err)
		}
		for i := 0; i < int(totalCallCount); i++ {
			key := fmt.Sprintf("key-%d", i)
			value := fmt.Sprintf("value-%d", i)
			it.MustValue(m.Put(context.Background(), key, value))
		}
		cc := atomic.LoadInt32(&callCount)
		if !assert.Equal(t, int32(0), cc) {
			t.Fatalf("call count target: %d, current: %d", totalCallCount, cc)
		}
```

Reply:
It's for making sure after removing the listener put operations do not fire events (hence remove listener is successful). What you have here seems correct to me, the counter doesn't increase and the test passes. But this line is misleading:
```
t.Fatalf("call count target: %d, current: %d", totalCallCount, cc)
```
It should be:
```
t.Fatalf("call count target: %d, current: %d", 0, cc)
```